reference/promise-types/measurements.markdown

```diff
@@ -20,27 +20,108 @@ bundle destined for the agent concerned, in this case `monitor`. However, it is
+
+Select all unique observables in the database:
+
+```sql
+SELECT distinct(observable) FROM monitoringmgmeta;
+```
+
+To see which measures are collected for each host, select all data from this
+table and then use report result filters to examine a particular host or measure's having
+data or not.
+
+```sql
+SELECT * FROM monitoringmgmeta;
+```
+
+More info at https://docs.cfengine.com/docs/3.18/reference-enterprise-api-ref-sql-schema.html#table-monitoringhg
+
+Monitoring data is presented in Mission Portal in two places:
+the Measurements App https://docs.cfengine.com/docs/3.18/enterprise-cfengine-guide-measurements.html
+and the Measurements section of the Host Info page https://docs.cfengine.com/docs/3.18/enterprise-cfengine-guide-hosts-health.html
+
+When policy is changed in regards to monitor bundles, `cf-monitord` should be restart as well as `cf-serverd` in order to receive the updated policy.
+
+
+It is possible to configure masterfiles to restart cf-monitord when variables which affect it's configuration are changed.
+
+https://docs.cfengine.com/docs/3.18/reference-masterfiles-policy-framework.html#configure-mpf-to-automatically-restart-components-on-relevant-data-change
+
+All measurements historical data is stored in `${sys.statedir}/cf_observations.lmdb`. This is where reporting data is pulled from.
+
+`${sys.statedir}/env_data` is a current snapshot of measurement values which agents (`cf-agent` and `cf-promises`) use to generate the special monitor variables.
+
+https://docs.cfengine.com/docs/master/reference-special-variables-mon.html
+
+`${sys.statedir}/ts_key` is the source for the list of observables/measurements used in the system. This file includes the integer id, name and other meta data in a slightly human readable form
+
+```
+0,users,Users with active processes - including system users,average users per 2.5 mins,0.000,100.000,1
+```
+
+The `ts_key` file should not be altered.
+
+Note that if a measurement has always had a value of zero it will not be reported and so not available in Mission Portal Measurements or Host Info pages.
+
 ```cf3
 bundle monitor self_watch
 {
   measurements:
     # Follow a special process over time
     # using CFEngine's process cache to avoid resampling
+
+    # Example content from /var/cfengine/state/cf_rootprocs
+    #USER                             PID         PPID          PGID          %CPU         %MEM        VSZ        NI         RSS    TTY      NLWP STIME     ELAPSED     TIME COMMAND
+    #root                             19103       1             19103         0.2          2.1         71716      0          10676  ?           1 18:09       40:13 00:00:06 /var/cfengine/bin/cf-monitord --no-fork
+
+    # match_value:
+    #root \s+                         [0-9.]+ \s+ [0-9.]+  \s+  [0-9.]+ \s+   [0-9.]+ \s+  [0-9.]+ \s+ [0-9]+ \s+ [0-9]+ \s+ ([0-9]+) .*"
+
   
      "/var/cfengine/state/cf_rootprocs"
   
         handle => "monitor_self_watch",
```

Comment:
I was thinking that we should rename this handle to better describe the actual measurement, e.g. `cf_monitord_RSS`